ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs

```diff
@@ -24,7 +25,8 @@ public async Task Invoke(HttpContext context)
         ILogEventEnricher[] enrichers =
         {
             new PropertyEnricher("deviceId", deviceId ?? ""),
-            new PropertyEnricher("identityAddress", identityAddress ?? "")
+            new PropertyEnricher("identityAddress", identityAddress ?? ""),
+            new PropertyEnricher("username", _userContext.GetUsernameOrNull() ?? context.GetOpenIddictServerRequest()?.Username)
```

Comment:
You forgot to handle null. Also, you should extract the username into a variable, just like deviceId and identityAddress
```suggestion
            var username = _userContext.GetUsernameOrNull() ?? context.GetOpenIddictServerRequest()?.Username;
            ...
            new PropertyEnricher("username", username ?? "")
```